### ISSUE TITLE
Find list of H2 users who have gone through ORCID integration

### ISSUE THREAD

**Issue by peetucket** [sul-dlss/happy-heron#3117]
Find a list of sunets that (1) exist in the H2 database as any kind of user and 2) have authorized Stanford to write to ORCID (using the MaIS API, which we have access to right now via a Ruby client in the sul-pub codebase).

**Comment by peetucket** [comment]
Fetch all ORCIDs from H2 works and write to CSV:

```
orcids = AbstractContributor.all.filter_map {|c| c.orcid unless c.orcid.blank?}.uniq!
CSV.open("tmp/h2_orcids.csv", "w") do |csv|
  csv << ['orcid']
  orcids.each {|orcid|  csv << [orcid]}
end
```

**Comment by peetucket** [comment]
[h2_orcids.csv](https://github.com/sul-dlss/happy-heron/files/11839544/h2_orcids.csv)

**Comment by peetucket** [comment]
Find list of users who have gone through integration and their scope (done on sul-pub-prod, which has the MaIS client):

```
FIND_SCOPE=true RAILS_ENV=production bundle exec rake sul:stanford_orcid_users['tmp/stanford_orcids.csv']
```

**Comment by peetucket** [comment]
Using both files:

```
require 'csv'
stanford_orcids = CSV.read('tmp/stanford_orcids.csv')
h2_orcids = CSV.parse(File.read("tmp/h2_orcids.csv"), headers: true)
stanford_orcids = CSV.parse(File.read("tmp/stanford_orcids.csv"), headers: true)

```

**Comment by peetucket** [comment]
@amyehodge I've collected some stats on ORCIDs that exist in H2 works and added them here.

**Comment by justinlittman** [comment]
I'd also suggest it would be helpful to:
* Remove library staff. (Or at least check a sample to determine what % are library staff.)
* Contact some of the users with write access to confirm that pushing SDR items to Orcid is expected and/or desired.

**Comment by peetucket** [comment]
See https://docs.google.com/spreadsheets/d/1csEtMDALUTW86avBisbQln-ypQeP12zi2vst4vVTW1Q/edit#gid=0 for full list of sunets who have both read and write scope defined.  Removing SUNETs from this ticket for privacy reasons.

There are some familiar library sunets in both lists unsurprisingly.

**Comment by amyehodge** [comment]
I am getting some excellent feedback from the read-only folks I emailed this morning. I included SUL folks just because their responses are easy enough to separate out. This is definitely prompting a number of these folks to update their permissions with write access. 👍 

I will compile the feedback once I think I have most of what I'll get, anonymize it, and put it in another sheet of the spreadsheet linked to above.
